File: android/src/main/java/com/easemob/im_flutter_sdk/EnumTools.java
package com.easemob.im_flutter_sdk;

import androidx.annotation.NonNull;

import com.hyphenate.chat.EMChatRoom.EMChatRoomPermissionType;
import com.hyphenate.chat.EMChatThreadEvent;
import com.hyphenate.chat.EMConversation;
import com.hyphenate.chat.EMFileMessageBody.EMDownloadStatus;
import com.hyphenate.chat.EMGroup;
import com.hyphenate.chat.EMMessage;
import com.hyphenate.chat.EMMessageReactionOperation;
import com.hyphenate.chat.EMPushManager;
import com.hyphenate.chat.EMSilentModeParam;


public class EnumTools {

    static int reactionOperationToInt(EMMessageReactionOperation.Operation operation) {
        return operation.ordinal();
    }

    static EMMessageReactionOperation.Operation reactionOperationFromInt(int iType) {
        return EMMessageReactionOperation.Operation.values()[iType];
    }

    static int threadOperationToInt(EMChatThreadEvent.TYPE type) {
        return type.ordinal();
    }

    static EMChatThreadEvent.TYPE threadOperationFromInt(int iType) {
        return EMChatThreadEvent.TYPE.values()[iType];
    }

    static int messageBodyTypeToInt(@NonNull EMMessage.Type type) {
        return type.ordinal();
    }

    static EMMessage.Direct messageDirectFromInt(int iType) {
        return EMMessage.Direct.values()[iType];
    }

    static int messageDirectToInt(@NonNull EMMessage.Direct direct) {
        return  direct.ordinal();
    }

    static EMMessage.Type messageBodyTypeFromInt(int iType) {
        return EMMessage.Type.values()[iType];
    }

    static int downloadStatusToInt(@NonNull EMDownloadStatus status) {
        return status.ordinal();
    }

    static EMDownloadStatus downloadStatusFromInt(int iStatus) {
        return  EMDownloadStatus.values()[iStatus];
    }

    static int chatTypeToInt(@NonNull EMMessage.ChatType type) {
        return  type.ordinal();
    }

    static EMMessage.ChatType chatTypeFromInt(int iType) {
        return EMMessage.ChatType.values()[iType];
    }

    static int chatRoomPermissionTypeToInt(@NonNull EMChatRoomPermissionType type) {
        return type.ordinal();
    }

    static EMChatRoomPermissionType chatRoomPermissionTypeFromInt(int iType) {
        return EMChatRoomPermissionType.values()[iType];
    }

    static int groupPermissionTypeToInt(@NonNull EMGroup.EMGroupPermissionType type) {
        return type.ordinal();
    }

    static EMGroup.EMGroupPermissionType groupPermissionTypeFromInt(int iType) {
        return EMGroup.EMGroupPermissionType.values()[iType];
    }


    static int searchDirectionToInt(EMConversation.EMSearchDirection direction) {
        return direction.ordinal();
    }

    static EMConversation.EMSearchDirection searchDirectionFromInt(int iDirection) {
        return  EMConversation.EMSearchDirection.values()[iDirection];
    }

    static EMConversation.EMConversationType conversationTypeFromInt(int iType) {
        return  EMConversation.EMConversationType.values()[iType];
    }

    static int conversationTypeToInt(EMConversation.EMConversationType type) {
        return type.ordinal();
    }

    static int silentModeParamTypeToInt(EMSilentModeParam.EMSilentModeParamType type) {
        return type.ordinal();
    }

    static EMSilentModeParam.EMSilentModeParamType silentModeParamTypeFromInt(int iType) {
        return EMSilentModeParam.EMSilentModeParamType.values()[iType];
    }


    static int remindTypeToInt(EMPushManager.EMPushRemindType type) {
        return type.ordinal();
    }

    static EMPushManager.EMPushRemindType remindTypeFromInt(int iType) {
        return EMPushManager.EMPushRemindType.values()[iType];
    }
    static EMMessage.Status messageStatusFromInt(int iStatus) {
        switch (iStatus) {
            case 0 :{
                return EMMessage.Status.CREATE;
            }
            case 1 :{
                return EMMessage.Status.INPROGRESS;
            }
            case 2 :{
                return EMMessage.Status.SUCCESS;
            }
            case 3 :{
                return EMMessage.Status.FAIL;
            }
        }
        return EMMessage.Status.FAIL;
    }

    static int messageStatusToInt(EMMessage.Status status) {
        switch (status){
            case CREATE:
                return 0;
            case INPROGRESS:
                return 1;
            case SUCCESS:
                return 2;
            case FAIL:
                return 3;
        }
        return 3;
    }
 }

File: ios/Classes/EnumTools.m
//
//  EnumTools.m
//  im_flutter_sdk
//
//  Created by 杜洁鹏 on 2024/10/23.
//

#import "EnumTools.h"

@implementation EnumTools
+(NSInteger)messageBodyTypeToInt:(EMMessageBodyType)value {
    switch (value) {
        case EMMessageBodyTypeText:
            return 0;
            break;
        case EMMessageBodyTypeImage:
            return 1;
            break;
        case EMMessageBodyTypeVideo:
            return 2;
            break;
        case EMMessageBodyTypeLocation:
            return 3;
            break;
        case EMMessageBodyTypeVoice:
            return 4;
            break;
        case EMMessageBodyTypeFile:
            return 5;
            break;
        case EMMessageBodyTypeCmd:
            return 6;
            break;
        case EMMessageBodyTypeCustom:
            return 7;
            break;
        case EMMessageBodyTypeCombine:
            return 8;
            break;
        default:
            break;
    }
    return 0;
}

+(EMMessageBodyType)messageBodyTypeFromInt:(NSInteger)value{
    switch (value) {
        case 0:
            return EMMessageBodyTypeText;
            break;
        case 1:
            return EMMessageBodyTypeImage;
            break;
        case 2:
            return EMMessageBodyTypeVideo;
            break;
        case 3:
            return EMMessageBodyTypeLocation;
            break;
        case 4:
            return EMMessageBodyTypeVoice;
            break;
        case 5:
            return EMMessageBodyTypeFile;
            break;
        case 6:
            return EMMessageBodyTypeCmd;
            break;
        case 7:
            return EMMessageBodyTypeCustom;
            break;
        case 8:
            return EMMessageBodyTypeCombine;
            break;
        default:
            break;
    }
    return EMMessageBodyTypeText;
}

+(NSInteger)downloadStatusToInt:(EMDownloadStatus)value{
    return value;
}

+(EMDownloadStatus)downloadStatusFromInt:(NSInteger)value{
    return value;
}

+(NSInteger)chatTypeToInt:(EMChatType)value{
    return value;
}

+(EMChatType)chatTypeFromInt:(NSInteger)value{
    return value;
}

+(NSInteger)chatRoomPermissionTypeToInt:(EMChatroomPermissionType)value{
    switch (value) {
        case EMChatroomPermissionTypeMember:
            return 0;
            break;
        case EMChatroomPermissionTypeAdmin:
            return 1;
            break;
        case EMChatroomPermissionTypeOwner:
            return 2;
            break;
        case EMChatroomPermissionTypeNone:
            return 3;
            break;
        default:
            break;
    }
    return 3;
}

+(EMChatroomPermissionType)chatRoomPermissionTypeFromInt:(NSInteger)value{
    switch (value) {
        case 0:
            return EMChatroomPermissionTypeMember;
            break;
        case 1:
            return EMChatroomPermissionTypeAdmin;
            break;
        case 2:
            return EMChatroomPermissionTypeOwner;
            break;
        case 3:
            return EMChatroomPermissionTypeNone;
            break;
        default:
            break;
    }
    return EMChatroomPermissionTypeNone;
}

+(NSInteger)groupPermissionTypeToInt:(EMGroupPermissionType)value{
    switch (value) {
        case EMGroupPermissionTypeMember:
            return 0;
            break;
        case EMGroupPermissionTypeAdmin:
            return 1;
            break;
        case EMGroupPermissionTypeOwner:
            return 2;
            break;
        case EMGroupPermissionTypeNone:
            return 3;
            break;
        default:
            break;
    }
    return 3;
}

+(EMGroupPermissionType)groupPermissionTypeFromInt:(NSInteger)value{
    switch (value) {
        case 0:
            return EMGroupPermissionTypeMember;
            break;
        case 1:
            return EMGroupPermissionTypeAdmin;
            break;
        case 2:
            return EMGroupPermissionTypeOwner;
            break;
        case 3:
            return EMGroupPermissionTypeNone;
            break;
        default:
            break;
    }
    return EMGroupPermissionTypeNone;
}

+(NSInteger)searchDirectionToInt:(EMMessageSearchDirection)value{
    return value;
}

+(EMMessageSearchDirection)searchDirectionFromInt:(NSInteger)value{
    return value;
}

+(NSInteger)conversationTypeToInt:(EMConversationType)value{
    return value;
}

+(EMConversationType)conversationTypeFromInt:(NSInteger)value{
    return value;
}

+(NSInteger)silentModeParamTypeToInt:(EMSilentModeParamType)value{
    return value;
}

+(EMSilentModeParamType)silentModeParamTypeFromInt:(NSInteger)value{
    return value;
}

+(NSInteger)remindTypeToInt:(EMPushRemindType)value{
    return value;
}

+(EMPushRemindType)remindTypeFromInt:(NSInteger)value{
    return value;
}

+(NSInteger)messageStatusToInt:(EMMessageStatus)value{
    return value;
}

+(EMMessageStatus)messageStatusFromInt:(NSInteger)value{
    return value;
}

+(NSInteger)messageDirectToInt:(EMMessageDirection)value {
    return value;
}

+(EMMessageDirection)messageDirectFromInt:(NSInteger)value {
    return value;
}

+(NSInteger)threadOperationToInt:(EMThreadOperation)value {
    return value;
}

+(EMThreadOperation)threadOperationFromInt:(NSInteger)value {
    return value;
}

+(NSInteger)reactionOperationToInt:(EMMessageReactionOperate)value {
    return value;
}

+(EMMessageReactionOperate)recationOperationFromInt:(NSInteger)value {
    return value;
}
@end
